### PR TITLE
Secure time split implementation and client interfaces.

### PR DESCRIPTION
**Pull request by alzix**
alternative implementation to PR #7 

naming convention is open for discussion.
the name `client` does not make sense for someone not familiar with SPM. 
Perhaps the word client can be replaced by Java like `ifs` for interface.

- secure_time_impl.c/h - real implementation
- secure_time_client.h - client interface
- secure_time_client_spe.h - client SPE only interface

- secure_time_client_direct.c - client interface direct API call implementation
- secure_time_client_spe_direct.c - SPE only client interface direct API call implementation

### CODE REVIEW

**Comment by romkuz01** [comment]
1) Within ..._impl functions there're some calls to other Secure Times API functions. These calls also should be changed to ..._impl function calls.
2) Why do we need to have client (or anything else) within the file name? I think it's simpler and less confusing to call files just secure_time.h/c, secure_time_spe.h/c, secure_time_[spe_]direct.c, secure_time_[spe_]proxy.c.

File: secure_time/secure_time_impl.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+#define __SECURE_TIME_IMPL_H__
+
+#include "secure_time_client_spe.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**< Maximum time a nonce is valid after generation.*/
+#define SECURE_TIME_NONCE_MAX_TIMEOUT_SECONDS           (300UL)
```
[the remark this answers is not in the document]
Comment (alzix) [reply secure_time/secure_time_impl.h]: this was moved from client header file as it is only used internally.

**Pull request merged**